be2-scala/src/test/scala/ch/epfl/pop/model/objects/SignatureSuite.scala

```diff
@@ -0,0 +1,99 @@
+package ch.epfl.pop.model.objects
+
+import org.scalatest.{FunSuite, Matchers, BeforeAndAfterAll}
+import com.google.crypto.tink.subtle.Ed25519Sign
+import java.nio.charset.StandardCharsets
+import org.scalatest.Inspectors.{forAll,forEvery}
+
+
+/*Helper object for testing*/
+case class TestObj(ed_signer: Ed25519Sign, keyPair: Ed25519Sign.KeyPair)
+class SignatureSuite extends FunSuite with Matchers with BeforeAndAfterAll{
+
+    var tester: TestObj = null;
+    override def beforeAll(){
+            val kpair = Ed25519Sign.KeyPair.newKeyPair()
+
+            val privateKey = kpair.getPrivateKey
+            val ed_signer  = new Ed25519Sign(privateKey)
+            tester = TestObj(ed_signer, kpair)
+    }
```

Comment:
I get what you are trying to do there. There are cleaner ways to do this without setting `tester` to `null` first. For example, try googling "Scalatest sharing fixtures"